{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "#with open('page.txt', 'r') as myfile:\n",
    " # background = myfile.read()\n",
    "# with open('page.txt','rb') as f:\n",
    "#     contents = f.read()\n",
    "\n",
    "background = u\"\"\"There are well developed protein subcellular localization prediction resources available, including protein subcellualr location databases, and prediction tools.[35][36]'b'Data from high-throughput chromosome conformation capture experiments, such as Hi-C (experiment) and ChIA-PET, can provide information on the spatial proximity of DNA loci. Analysis of these experiments can determine the three-dimensional structure and nuclear organization of chromatin. Bioinformatic challenges in this field include partitioning the genome into domains, such as Topologically Associating Domains (TADs), that are organised together in three-dimensional space.[37]'b'Protein structure prediction is another important application of bioinformatics. The amino acid sequence of a protein, the so-called primary structure, can be easily determined from the sequence on the gene that codes for it. In the vast majority of cases, this primary structure uniquely determines a structure in its native environment. (Of course, there are exceptions, such as the bovine spongiform encephalopathy \\xe2\\x80\\x93 a.k.a. Mad Cow Disease \\xe2\\x80\\x93 prion.) Knowledge of this structure is vital in understanding the function of the protein. Structural information is usually classified as one of secondary, tertiary and quaternary structure. A viable general solution to such predictions remains an open problem. Most efforts have so far been directed towards heuristics that work most of the time.[citation needed]'b\"One of the key ideas in bioinformatics is the notion of homology. In the genomic branch of bioinformatics, homology is used to predict the function of a gene: if the sequence of gene A, whose function is known, is homologous to the sequence of gene B, whose function is unknown, one could infer that B may share A's function. In the structural branch of bioinformatics, homology is used to determine which parts of a protein are important in structure formation and interaction with other proteins. In a technique called homology modeling, this information is used to predict the structure of a protein once the structure of a homologous protein is known. This currently remains the only way to predict protein structures reliably.\"b'One example of this is the similar protein homology between hemoglobin in humans and the hemoglobin in legumes (leghemoglobin). Both serve the same purpose of transporting oxygen in the organism. Though both of these proteins have completely different amino acid sequences, their protein structures are virtually identical, which reflects their near identical purposes.[38]'b'Other techniques for predicting protein structure include protein threading and de novo (from scratch) physics-based modeling.'b'Network analysis seeks to understand the relationships within biological networks such as metabolic or protein\\xe2\\x80\\x93protein interaction networks. Although biological networks can be constructed from a single type of molecule or entity (such as genes), network biology often attempts to integrate many different data types, such as proteins, small molecules, gene expression data, and others, which are all connected physically, functionally, or both.'b'Systems biology involves the use of computer simulations of cellular subsystems (such as the networks of metabolites and enzymes that comprise metabolism, signal transduction pathways and gene regulatory networks) to both analyze and visualize the complex connections of these cellular processes. Artificial life or virtual evolution attempts to understand evolutionary processes via the computer simulation of simple (artificial) life forms.'b'Tens of thousands of three-dimensional protein structures have been determined by X-ray crystallography and protein nuclear magnetic resonance spectroscopy (protein NMR) and a central question in structural bioinformatics is whether it is practical to predict possible protein\\xe2\\x80\\x93protein interactions only based on these 3D shapes, without performing protein\\xe2\\x80\\x93protein interaction experiments. A variety of methods have been developed to tackle the protein\\xe2\\x80\\x93protein docking problem, though it seems that there is still much work to be done in this field.'b'Other interactions encountered in the field include Protein\\xe2\\x80\\x93ligand (including drug) and protein\\xe2\\x80\\x93peptide. Molecular dynamic simulation of movement of atoms about rotatable bonds is the fundamental principle behind computational algorithms, termed docking algorithms, for studying molecular interactions.'b'The growth in the number of published literature makes it virtually impossible to read every paper, resulting in disjointed sub-fields of research. Literature analysis aims to employ computational and statistical linguistics to mine this growing library of text resources. For example:'b'The area of research draws from statistics and computational linguistics.'b\"Computational technologies are used to accelerate or fully automate the processing, quantification and analysis of large amounts of high-information-content biomedical imagery. Modern image analysis systems augment an observer's ability to make measurements from a large or complex set of images, by improving accuracy, objectivity, or speed. A fully developed analysis system may completely replace the observer. Although these systems are not unique to biomedical imagery, biomedical imaging is becoming more important for both diagnostics and research. Some examples are:\"b'Computational techniques are used to analyse high-throughput, low-measurement single cell data, such as that obtained from flow cytometry. These methods typically involve finding populations of cells that are relevant to a particular disease state or experimental condition.'b'Biodiversity informatics deals with the collection and analysis of biodiversity data, such as taxonomic databases, or microbiome data. Examples of such analyses include phylogenetics, niche modelling, species richness mapping, DNA barcoding, or species identification tools.'b'Biological ontologies are directed acyclic graphs of controlled vocabularies. They are designed to capture biological concepts and descriptions in a way that can be easily categorised and analysed with computers. When categorised in this way, it is possible to gain added value from holistic and integrated analysis.'b'The OBO Foundry was an effort to standardise certain ontologies. One of the most widespread is the Gene ontology which describes gene function. There are also ontologies which describe phenotypes.'b'Databases are essential for bioinformatics research and applications. Many databases exist, covering various information types: for example, DNA and protein sequences, molecular structures, phenotypes and biodiversity. Databases may contain empirical data (obtained directly from experiments), predicted data (obtained from analysis), or, most commonly, both.\"\"\" \n",
    "background = background.encode('ascii',errors='ignore').decode()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['There are well developed protein subcellular localization prediction resources available, including protein subcellualr location databases, and prediction tools',\n",
       " \"[35][36]'b'Data from high-throughput chromosome conformation capture experiments, such as Hi-C (experiment) and ChIA-PET, can provide information on the spatial proximity of DNA loci\",\n",
       " ' Analysis of these experiments can determine the three-dimensional structure and nuclear organization of chromatin',\n",
       " ' Bioinformatic challenges in this field include partitioning the genome into domains, such as Topologically Associating Domains (TADs), that are organised together in three-dimensional space',\n",
       " \"[37]'b'Protein structure prediction is another important application of bioinformatics\",\n",
       " ' The amino acid sequence of a protein, the so-called primary structure, can be easily determined from the sequence on the gene that codes for it',\n",
       " ' In the vast majority of cases, this primary structure uniquely determines a structure in its native environment',\n",
       " ' (Of course, there are exceptions, such as the bovine spongiform encephalopathy  a',\n",
       " 'k',\n",
       " 'a',\n",
       " ' Mad Cow Disease  prion',\n",
       " ') Knowledge of this structure is vital in understanding the function of the protein',\n",
       " ' Structural information is usually classified as one of secondary, tertiary and quaternary structure',\n",
       " ' A viable general solution to such predictions remains an open problem',\n",
       " ' Most efforts have so far been directed towards heuristics that work most of the time',\n",
       " '[citation needed]\\'b\"One of the key ideas in bioinformatics is the notion of homology',\n",
       " \" In the genomic branch of bioinformatics, homology is used to predict the function of a gene: if the sequence of gene A, whose function is known, is homologous to the sequence of gene B, whose function is unknown, one could infer that B may share A's function\",\n",
       " ' In the structural branch of bioinformatics, homology is used to determine which parts of a protein are important in structure formation and interaction with other proteins',\n",
       " ' In a technique called homology modeling, this information is used to predict the structure of a protein once the structure of a homologous protein is known',\n",
       " ' This currently remains the only way to predict protein structures reliably',\n",
       " '\"b\\'One example of this is the similar protein homology between hemoglobin in humans and the hemoglobin in legumes (leghemoglobin)',\n",
       " ' Both serve the same purpose of transporting oxygen in the organism',\n",
       " ' Though both of these proteins have completely different amino acid sequences, their protein structures are virtually identical, which reflects their near identical purposes',\n",
       " \"[38]'b'Other techniques for predicting protein structure include protein threading and de novo (from scratch) physics-based modeling\",\n",
       " \"'b'Network analysis seeks to understand the relationships within biological networks such as metabolic or proteinprotein interaction networks\",\n",
       " ' Although biological networks can be constructed from a single type of molecule or entity (such as genes), network biology often attempts to integrate many different data types, such as proteins, small molecules, gene expression data, and others, which are all connected physically, functionally, or both',\n",
       " \"'b'Systems biology involves the use of computer simulations of cellular subsystems (such as the networks of metabolites and enzymes that comprise metabolism, signal transduction pathways and gene regulatory networks) to both analyze and visualize the complex connections of these cellular processes\",\n",
       " ' Artificial life or virtual evolution attempts to understand evolutionary processes via the computer simulation of simple (artificial) life forms',\n",
       " \"'b'Tens of thousands of three-dimensional protein structures have been determined by X-ray crystallography and protein nuclear magnetic resonance spectroscopy (protein NMR) and a central question in structural bioinformatics is whether it is practical to predict possible proteinprotein interactions only based on these 3D shapes, without performing proteinprotein interaction experiments\",\n",
       " ' A variety of methods have been developed to tackle the proteinprotein docking problem, though it seems that there is still much work to be done in this field',\n",
       " \"'b'Other interactions encountered in the field include Proteinligand (including drug) and proteinpeptide\",\n",
       " ' Molecular dynamic simulation of movement of atoms about rotatable bonds is the fundamental principle behind computational algorithms, termed docking algorithms, for studying molecular interactions',\n",
       " \"'b'The growth in the number of published literature makes it virtually impossible to read every paper, resulting in disjointed sub-fields of research\",\n",
       " ' Literature analysis aims to employ computational and statistical linguistics to mine this growing library of text resources',\n",
       " \" For example:'b'The area of research draws from statistics and computational linguistics\",\n",
       " '\\'b\"Computational technologies are used to accelerate or fully automate the processing, quantification and analysis of large amounts of high-information-content biomedical imagery',\n",
       " \" Modern image analysis systems augment an observer's ability to make measurements from a large or complex set of images, by improving accuracy, objectivity, or speed\",\n",
       " ' A fully developed analysis system may completely replace the observer',\n",
       " ' Although these systems are not unique to biomedical imagery, biomedical imaging is becoming more important for both diagnostics and research',\n",
       " ' Some examples are:\"b\\'Computational techniques are used to analyse high-throughput, low-measurement single cell data, such as that obtained from flow cytometry',\n",
       " ' These methods typically involve finding populations of cells that are relevant to a particular disease state or experimental condition',\n",
       " \"'b'Biodiversity informatics deals with the collection and analysis of biodiversity data, such as taxonomic databases, or microbiome data\",\n",
       " ' Examples of such analyses include phylogenetics, niche modelling, species richness mapping, DNA barcoding, or species identification tools',\n",
       " \"'b'Biological ontologies are directed acyclic graphs of controlled vocabularies\",\n",
       " ' They are designed to capture biological concepts and descriptions in a way that can be easily categorised and analysed with computers',\n",
       " ' When categorised in this way, it is possible to gain added value from holistic and integrated analysis',\n",
       " \"'b'The OBO Foundry was an effort to standardise certain ontologies\",\n",
       " ' One of the most widespread is the Gene ontology which describes gene function',\n",
       " ' There are also ontologies which describe phenotypes',\n",
       " \"'b'Databases are essential for bioinformatics research and applications\",\n",
       " ' Many databases exist, covering various information types: for example, DNA and protein sequences, molecular structures, phenotypes and biodiversity',\n",
       " ' Databases may contain empirical data (obtained directly from experiments), predicted data (obtained from analysis), or, most commonly, both',\n",
       " '']"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "background.split('.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['[35]', '[36]', '[37]', '[38]']"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "re.findall(r\"\\[\\d+\\]\", background)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "background_nofoot = re.sub(r\"\\[\\d+\\]\", \"\", background)\n",
    "#background_nofoot = re.sub(r'/[\\x00-\\x1F\\x7F\\xA0]/u',r'', background_nofoot) \n",
    "#background_nofoot.decode(\"utf-8\").encode(\"ascii\", \"ignore\")\n",
    "#background_nofoot = re.sub(r'[^\\x00-\\xff]','',background_nofoot)\n",
    "\n",
    "splitter = re.compile(r\"\"\"\n",
    "    [.!?]       # split on punctuation\n",
    "    \"\"\", re.VERBOSE)\n",
    "\n",
    "splitter = re.compile(r\"\"\"\n",
    "    (?<![A-Z])  # last character cannot be uppercase\n",
    "    [.!?]       # match punctuation\n",
    "    \\s+         # followed by whitespace\n",
    "    (?=[A-Z])   # next character must be uppercase\n",
    "    \"\"\", re.VERBOSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are well developed protein subcellular localization prediction resources available, including protein subcellualr location databases, and prediction tools.'b'Data from high-throughput chromosome conformation capture experiments, such as Hi-C (experiment) and ChIA-PET, can provide information on the spatial proximity of DNA loci\n",
      "--\n",
      "Analysis of these experiments can determine the three-dimensional structure and nuclear organization of chromatin\n",
      "--\n",
      "Bioinformatic challenges in this field include partitioning the genome into domains, such as Topologically Associating Domains (TADs), that are organised together in three-dimensional space.'b'Protein structure prediction is another important application of bioinformatics\n",
      "--\n",
      "The amino acid sequence of a protein, the so-called primary structure, can be easily determined from the sequence on the gene that codes for it\n",
      "--\n",
      "In the vast majority of cases, this primary structure uniquely determines a structure in its native environment. (Of course, there are exceptions, such as the bovine spongiform encephalopathy  a.k.a\n",
      "--\n",
      "Mad Cow Disease  prion.) Knowledge of this structure is vital in understanding the function of the protein\n",
      "--\n",
      "Structural information is usually classified as one of secondary, tertiary and quaternary structure\n",
      "--\n",
      "A viable general solution to such predictions remains an open problem\n",
      "--\n",
      "Most efforts have so far been directed towards heuristics that work most of the time.[citation needed]'b\"One of the key ideas in bioinformatics is the notion of homology\n",
      "--\n",
      "In the genomic branch of bioinformatics, homology is used to predict the function of a gene: if the sequence of gene A, whose function is known, is homologous to the sequence of gene B, whose function is unknown, one could infer that B may share A's function\n",
      "--\n",
      "In the structural branch of bioinformatics, homology is used to determine which parts of a protein are important in structure formation and interaction with other proteins\n",
      "--\n",
      "In a technique called homology modeling, this information is used to predict the structure of a protein once the structure of a homologous protein is known\n",
      "--\n",
      "This currently remains the only way to predict protein structures reliably.\"b'One example of this is the similar protein homology between hemoglobin in humans and the hemoglobin in legumes (leghemoglobin)\n",
      "--\n",
      "Both serve the same purpose of transporting oxygen in the organism\n",
      "--\n",
      "Though both of these proteins have completely different amino acid sequences, their protein structures are virtually identical, which reflects their near identical purposes.'b'Other techniques for predicting protein structure include protein threading and de novo (from scratch) physics-based modeling.'b'Network analysis seeks to understand the relationships within biological networks such as metabolic or proteinprotein interaction networks\n",
      "--\n",
      "Although biological networks can be constructed from a single type of molecule or entity (such as genes), network biology often attempts to integrate many different data types, such as proteins, small molecules, gene expression data, and others, which are all connected physically, functionally, or both.'b'Systems biology involves the use of computer simulations of cellular subsystems (such as the networks of metabolites and enzymes that comprise metabolism, signal transduction pathways and gene regulatory networks) to both analyze and visualize the complex connections of these cellular processes\n",
      "--\n",
      "Artificial life or virtual evolution attempts to understand evolutionary processes via the computer simulation of simple (artificial) life forms.'b'Tens of thousands of three-dimensional protein structures have been determined by X-ray crystallography and protein nuclear magnetic resonance spectroscopy (protein NMR) and a central question in structural bioinformatics is whether it is practical to predict possible proteinprotein interactions only based on these 3D shapes, without performing proteinprotein interaction experiments\n",
      "--\n",
      "A variety of methods have been developed to tackle the proteinprotein docking problem, though it seems that there is still much work to be done in this field.'b'Other interactions encountered in the field include Proteinligand (including drug) and proteinpeptide\n",
      "--\n",
      "Molecular dynamic simulation of movement of atoms about rotatable bonds is the fundamental principle behind computational algorithms, termed docking algorithms, for studying molecular interactions.'b'The growth in the number of published literature makes it virtually impossible to read every paper, resulting in disjointed sub-fields of research\n",
      "--\n",
      "Literature analysis aims to employ computational and statistical linguistics to mine this growing library of text resources\n",
      "--\n",
      "For example:'b'The area of research draws from statistics and computational linguistics.'b\"Computational technologies are used to accelerate or fully automate the processing, quantification and analysis of large amounts of high-information-content biomedical imagery\n",
      "--\n",
      "Modern image analysis systems augment an observer's ability to make measurements from a large or complex set of images, by improving accuracy, objectivity, or speed\n",
      "--\n",
      "A fully developed analysis system may completely replace the observer\n",
      "--\n",
      "Although these systems are not unique to biomedical imagery, biomedical imaging is becoming more important for both diagnostics and research\n",
      "--\n",
      "Some examples are:\"b'Computational techniques are used to analyse high-throughput, low-measurement single cell data, such as that obtained from flow cytometry\n",
      "--\n",
      "These methods typically involve finding populations of cells that are relevant to a particular disease state or experimental condition.'b'Biodiversity informatics deals with the collection and analysis of biodiversity data, such as taxonomic databases, or microbiome data\n",
      "--\n",
      "Examples of such analyses include phylogenetics, niche modelling, species richness mapping, DNA barcoding, or species identification tools.'b'Biological ontologies are directed acyclic graphs of controlled vocabularies\n",
      "--\n",
      "They are designed to capture biological concepts and descriptions in a way that can be easily categorised and analysed with computers\n",
      "--\n",
      "When categorised in this way, it is possible to gain added value from holistic and integrated analysis.'b'The OBO Foundry was an effort to standardise certain ontologies\n",
      "--\n",
      "One of the most widespread is the Gene ontology which describes gene function\n",
      "--\n",
      "There are also ontologies which describe phenotypes.'b'Databases are essential for bioinformatics research and applications\n",
      "--\n",
      "Many databases exist, covering various information types: for example, DNA and protein sequences, molecular structures, phenotypes and biodiversity\n",
      "--\n",
      "Databases may contain empirical data (obtained directly from experiments), predicted data (obtained from analysis), or, most commonly, both.\n",
      "--\n"
     ]
    }
   ],
   "source": [
    "for sentence in splitter.split(background_nofoot):\n",
    "    print(sentence.strip())\n",
    "    print(\"--\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "word_splitter = re.compile(r\"\"\"\n",
    "    (\\w+)\n",
    "    \"\"\", re.VERBOSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sent_words = [word_splitter.findall(sent)\n",
    "              for sent in splitter.split(background_nofoot)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sent_words_lower = [[w.lower() for w in sent]\n",
    "                    for sent in sent_words]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "33"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sent_words_lower)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "allwords=[w for sent in sent_words_lower for w in sent]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['3d',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'a',\n",
       " 'ability',\n",
       " 'about',\n",
       " 'accelerate',\n",
       " 'accuracy',\n",
       " 'acid',\n",
       " 'acid',\n",
       " 'acyclic',\n",
       " 'added',\n",
       " 'aims',\n",
       " 'algorithms',\n",
       " 'algorithms',\n",
       " 'all',\n",
       " 'also',\n",
       " 'although',\n",
       " 'although',\n",
       " 'amino',\n",
       " 'amino',\n",
       " 'amounts',\n",
       " 'an',\n",
       " 'an',\n",
       " 'an',\n",
       " 'analyse',\n",
       " 'analysed',\n",
       " 'analyses',\n",
       " 'analysis',\n",
       " 'analysis',\n",
       " 'analysis',\n",
       " 'analysis',\n",
       " 'analysis',\n",
       " 'analysis',\n",
       " 'analysis',\n",
       " 'analysis',\n",
       " 'analysis',\n",
       " 'analyze',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'and',\n",
       " 'another',\n",
       " 'application',\n",
       " 'applications',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'are',\n",
       " 'area',\n",
       " 'artificial',\n",
       " 'artificial',\n",
       " 'as',\n",
       " 'as',\n",
       " 'as',\n",
       " 'as',\n",
       " 'as',\n",
       " 'as',\n",
       " 'as',\n",
       " 'as',\n",
       " 'as',\n",
       " 'as',\n",
       " 'associating',\n",
       " 'atoms',\n",
       " 'attempts',\n",
       " 'attempts',\n",
       " 'augment',\n",
       " 'automate',\n",
       " 'available',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'barcoding',\n",
       " 'based',\n",
       " 'based',\n",
       " 'be',\n",
       " 'be',\n",
       " 'be',\n",
       " 'be',\n",
       " 'becoming',\n",
       " 'been',\n",
       " 'been',\n",
       " 'been',\n",
       " 'behind',\n",
       " 'between',\n",
       " 'biodiversity',\n",
       " 'biodiversity',\n",
       " 'biodiversity',\n",
       " 'bioinformatic',\n",
       " 'bioinformatics',\n",
       " 'bioinformatics',\n",
       " 'bioinformatics',\n",
       " 'bioinformatics',\n",
       " 'bioinformatics',\n",
       " 'bioinformatics',\n",
       " 'biological',\n",
       " 'biological',\n",
       " 'biological',\n",
       " 'biological',\n",
       " 'biology',\n",
       " 'biology',\n",
       " 'biomedical',\n",
       " 'biomedical',\n",
       " 'biomedical',\n",
       " 'bonds',\n",
       " 'both',\n",
       " 'both',\n",
       " 'both',\n",
       " 'both',\n",
       " 'both',\n",
       " 'both',\n",
       " 'bovine',\n",
       " 'branch',\n",
       " 'branch',\n",
       " 'by',\n",
       " 'by',\n",
       " 'c',\n",
       " 'called',\n",
       " 'called',\n",
       " 'can',\n",
       " 'can',\n",
       " 'can',\n",
       " 'can',\n",
       " 'can',\n",
       " 'capture',\n",
       " 'capture',\n",
       " 'cases',\n",
       " 'categorised',\n",
       " 'categorised',\n",
       " 'cell',\n",
       " 'cells',\n",
       " 'cellular',\n",
       " 'cellular',\n",
       " 'central',\n",
       " 'certain',\n",
       " 'challenges',\n",
       " 'chia',\n",
       " 'chromatin',\n",
       " 'chromosome',\n",
       " 'citation',\n",
       " 'classified',\n",
       " 'codes',\n",
       " 'collection',\n",
       " 'commonly',\n",
       " 'completely',\n",
       " 'completely',\n",
       " 'complex',\n",
       " 'complex',\n",
       " 'comprise',\n",
       " 'computational',\n",
       " 'computational',\n",
       " 'computational',\n",
       " 'computational',\n",
       " 'computational',\n",
       " 'computer',\n",
       " 'computer',\n",
       " 'computers',\n",
       " 'concepts',\n",
       " 'condition',\n",
       " 'conformation',\n",
       " 'connected',\n",
       " 'connections',\n",
       " 'constructed',\n",
       " 'contain',\n",
       " 'content',\n",
       " 'controlled',\n",
       " 'could',\n",
       " 'course',\n",
       " 'covering',\n",
       " 'cow',\n",
       " 'crystallography',\n",
       " 'currently',\n",
       " 'cytometry',\n",
       " 'data',\n",
       " 'data',\n",
       " 'data',\n",
       " 'data',\n",
       " 'data',\n",
       " 'data',\n",
       " 'data',\n",
       " 'data',\n",
       " 'databases',\n",
       " 'databases',\n",
       " 'databases',\n",
       " 'databases',\n",
       " 'databases',\n",
       " 'de',\n",
       " 'deals',\n",
       " 'describe',\n",
       " 'describes',\n",
       " 'descriptions',\n",
       " 'designed',\n",
       " 'determine',\n",
       " 'determine',\n",
       " 'determined',\n",
       " 'determined',\n",
       " 'determines',\n",
       " 'developed',\n",
       " 'developed',\n",
       " 'developed',\n",
       " 'diagnostics',\n",
       " 'different',\n",
       " 'different',\n",
       " 'dimensional',\n",
       " 'dimensional',\n",
       " 'dimensional',\n",
       " 'directed',\n",
       " 'directed',\n",
       " 'directly',\n",
       " 'disease',\n",
       " 'disease',\n",
       " 'disjointed',\n",
       " 'dna',\n",
       " 'dna',\n",
       " 'dna',\n",
       " 'docking',\n",
       " 'docking',\n",
       " 'domains',\n",
       " 'domains',\n",
       " 'done',\n",
       " 'draws',\n",
       " 'drug',\n",
       " 'dynamic',\n",
       " 'easily',\n",
       " 'easily',\n",
       " 'effort',\n",
       " 'efforts',\n",
       " 'empirical',\n",
       " 'employ',\n",
       " 'encephalopathy',\n",
       " 'encountered',\n",
       " 'entity',\n",
       " 'environment',\n",
       " 'enzymes',\n",
       " 'essential',\n",
       " 'every',\n",
       " 'evolution',\n",
       " 'evolutionary',\n",
       " 'example',\n",
       " 'example',\n",
       " 'example',\n",
       " 'examples',\n",
       " 'examples',\n",
       " 'exceptions',\n",
       " 'exist',\n",
       " 'experiment',\n",
       " 'experimental',\n",
       " 'experiments',\n",
       " 'experiments',\n",
       " 'experiments',\n",
       " 'experiments',\n",
       " 'expression',\n",
       " 'far',\n",
       " 'field',\n",
       " 'field',\n",
       " 'field',\n",
       " 'fields',\n",
       " 'finding',\n",
       " 'flow',\n",
       " 'for',\n",
       " 'for',\n",
       " 'for',\n",
       " 'for',\n",
       " 'for',\n",
       " 'for',\n",
       " 'for',\n",
       " 'formation',\n",
       " 'forms',\n",
       " 'foundry',\n",
       " 'from',\n",
       " 'from',\n",
       " 'from',\n",
       " 'from',\n",
       " 'from',\n",
       " 'from',\n",
       " 'from',\n",
       " 'from',\n",
       " 'from',\n",
       " 'from',\n",
       " 'fully',\n",
       " 'fully',\n",
       " 'function',\n",
       " 'function',\n",
       " 'function',\n",
       " 'function',\n",
       " 'function',\n",
       " 'function',\n",
       " 'functionally',\n",
       " 'fundamental',\n",
       " 'gain',\n",
       " 'gene',\n",
       " 'gene',\n",
       " 'gene',\n",
       " 'gene',\n",
       " 'gene',\n",
       " 'gene',\n",
       " 'gene',\n",
       " 'gene',\n",
       " 'general',\n",
       " 'genes',\n",
       " 'genome',\n",
       " 'genomic',\n",
       " 'graphs',\n",
       " 'growing',\n",
       " 'growth',\n",
       " 'have',\n",
       " 'have',\n",
       " 'have',\n",
       " 'have',\n",
       " 'hemoglobin',\n",
       " 'hemoglobin',\n",
       " 'heuristics',\n",
       " 'hi',\n",
       " 'high',\n",
       " 'high',\n",
       " 'high',\n",
       " 'holistic',\n",
       " 'homologous',\n",
       " 'homologous',\n",
       " 'homology',\n",
       " 'homology',\n",
       " 'homology',\n",
       " 'homology',\n",
       " 'homology',\n",
       " 'humans',\n",
       " 'ideas',\n",
       " 'identical',\n",
       " 'identical',\n",
       " 'identification',\n",
       " 'if',\n",
       " 'image',\n",
       " 'imagery',\n",
       " 'imagery',\n",
       " 'images',\n",
       " 'imaging',\n",
       " 'important',\n",
       " 'important',\n",
       " 'important',\n",
       " 'impossible',\n",
       " 'improving',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'in',\n",
       " 'include',\n",
       " 'include',\n",
       " 'include',\n",
       " 'include',\n",
       " 'including',\n",
       " 'including',\n",
       " 'infer',\n",
       " 'informatics',\n",
       " 'information',\n",
       " 'information',\n",
       " 'information',\n",
       " 'information',\n",
       " 'information',\n",
       " 'integrate',\n",
       " 'integrated',\n",
       " 'interaction',\n",
       " 'interaction',\n",
       " 'interaction',\n",
       " 'interactions',\n",
       " 'interactions',\n",
       " 'interactions',\n",
       " 'into',\n",
       " 'involve',\n",
       " 'involves',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'is',\n",
       " 'it',\n",
       " 'it',\n",
       " 'it',\n",
       " 'it',\n",
       " 'it',\n",
       " 'its',\n",
       " 'k',\n",
       " 'key',\n",
       " 'knowledge',\n",
       " 'known',\n",
       " 'known',\n",
       " 'large',\n",
       " 'large',\n",
       " 'leghemoglobin',\n",
       " 'legumes',\n",
       " 'library',\n",
       " 'life',\n",
       " 'life',\n",
       " 'linguistics',\n",
       " 'linguistics',\n",
       " 'literature',\n",
       " 'literature',\n",
       " 'localization',\n",
       " 'location',\n",
       " 'loci',\n",
       " 'low',\n",
       " 'mad',\n",
       " 'magnetic',\n",
       " 'majority',\n",
       " 'make',\n",
       " 'makes',\n",
       " 'many',\n",
       " 'many',\n",
       " 'mapping',\n",
       " 'may',\n",
       " 'may',\n",
       " 'may',\n",
       " 'measurement',\n",
       " 'measurements',\n",
       " 'metabolic',\n",
       " 'metabolism',\n",
       " 'metabolites',\n",
       " 'methods',\n",
       " 'methods',\n",
       " 'microbiome',\n",
       " 'mine',\n",
       " 'modeling',\n",
       " 'modeling',\n",
       " 'modelling',\n",
       " 'modern',\n",
       " 'molecular',\n",
       " 'molecular',\n",
       " 'molecular',\n",
       " 'molecule',\n",
       " 'molecules',\n",
       " 'more',\n",
       " 'most',\n",
       " 'most',\n",
       " 'most',\n",
       " 'most',\n",
       " 'movement',\n",
       " 'much',\n",
       " 'native',\n",
       " 'near',\n",
       " 'needed',\n",
       " 'network',\n",
       " 'network',\n",
       " 'networks',\n",
       " 'networks',\n",
       " 'networks',\n",
       " 'networks',\n",
       " 'networks',\n",
       " 'niche',\n",
       " 'nmr',\n",
       " 'not',\n",
       " 'notion',\n",
       " 'novo',\n",
       " 'nuclear',\n",
       " 'nuclear',\n",
       " 'number',\n",
       " 'objectivity',\n",
       " 'obo',\n",
       " 'observer',\n",
       " 'observer',\n",
       " 'obtained',\n",
       " 'obtained',\n",
       " 'obtained',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'of',\n",
       " 'often',\n",
       " 'on',\n",
       " 'on',\n",
       " 'on',\n",
       " 'once',\n",
       " 'one',\n",
       " 'one',\n",
       " 'one',\n",
       " 'one',\n",
       " 'one',\n",
       " 'only',\n",
       " 'only',\n",
       " 'ontologies',\n",
       " 'ontologies',\n",
       " 'ontologies',\n",
       " 'ontology',\n",
       " 'open',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'or',\n",
       " 'organised',\n",
       " 'organism',\n",
       " 'organization',\n",
       " 'other',\n",
       " 'other',\n",
       " 'other',\n",
       " 'others',\n",
       " 'oxygen',\n",
       " 'paper',\n",
       " 'particular',\n",
       " 'partitioning',\n",
       " 'parts',\n",
       " 'pathways',\n",
       " 'performing',\n",
       " 'pet',\n",
       " 'phenotypes',\n",
       " 'phenotypes',\n",
       " 'phylogenetics',\n",
       " 'physically',\n",
       " 'physics',\n",
       " 'populations',\n",
       " 'possible',\n",
       " 'possible',\n",
       " 'practical',\n",
       " 'predict',\n",
       " 'predict',\n",
       " 'predict',\n",
       " 'predict',\n",
       " 'predicted',\n",
       " 'predicting',\n",
       " 'prediction',\n",
       " 'prediction',\n",
       " 'prediction',\n",
       " 'predictions',\n",
       " 'primary',\n",
       " 'primary',\n",
       " 'principle',\n",
       " 'prion',\n",
       " 'problem',\n",
       " 'problem',\n",
       " 'processes',\n",
       " 'processes',\n",
       " 'processing',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'protein',\n",
       " 'proteinligand',\n",
       " 'proteinpeptide',\n",
       " 'proteinprotein',\n",
       " 'proteinprotein',\n",
       " 'proteinprotein',\n",
       " 'proteinprotein',\n",
       " 'proteins',\n",
       " 'proteins',\n",
       " 'proteins',\n",
       " 'provide',\n",
       " 'proximity',\n",
       " 'published',\n",
       " 'purpose',\n",
       " 'purposes',\n",
       " 'quantification',\n",
       " 'quaternary',\n",
       " 'question',\n",
       " 'ray',\n",
       " 'read',\n",
       " 'reflects',\n",
       " 'regulatory',\n",
       " 'relationships',\n",
       " 'relevant',\n",
       " 'reliably',\n",
       " 'remains',\n",
       " 'remains',\n",
       " 'replace',\n",
       " 'research',\n",
       " 'research',\n",
       " 'research',\n",
       " 'research',\n",
       " 'resonance',\n",
       " 'resources',\n",
       " 'resources',\n",
       " 'resulting',\n",
       " 'richness',\n",
       " 'rotatable',\n",
       " 's',\n",
       " 's',\n",
       " 'same',\n",
       " 'scratch',\n",
       " 'secondary',\n",
       " 'seeks',\n",
       " 'seems',\n",
       " 'sequence',\n",
       " 'sequence',\n",
       " 'sequence',\n",
       " 'sequence',\n",
       " 'sequences',\n",
       " 'sequences',\n",
       " 'serve',\n",
       " 'set',\n",
       " 'shapes',\n",
       " 'share',\n",
       " 'signal',\n",
       " 'similar',\n",
       " 'simple',\n",
       " 'simulation',\n",
       " 'simulation',\n",
       " 'simulations',\n",
       " 'single',\n",
       " 'single',\n",
       " 'small',\n",
       " 'so',\n",
       " 'so',\n",
       " 'solution',\n",
       " 'some',\n",
       " 'space',\n",
       " 'spatial',\n",
       " 'species',\n",
       " 'species',\n",
       " 'spectroscopy',\n",
       " 'speed',\n",
       " 'spongiform',\n",
       " 'standardise',\n",
       " 'state',\n",
       " 'statistical',\n",
       " 'statistics',\n",
       " 'still',\n",
       " 'structural',\n",
       " 'structural',\n",
       " 'structural',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structure',\n",
       " 'structures',\n",
       " 'structures',\n",
       " 'structures',\n",
       " 'structures',\n",
       " 'studying',\n",
       " 'sub',\n",
       " 'subcellualr',\n",
       " 'subcellular',\n",
       " 'subsystems',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'such',\n",
       " 'system',\n",
       " 'systems',\n",
       " 'systems',\n",
       " 'systems',\n",
       " 'tackle',\n",
       " 'tads',\n",
       " 'taxonomic',\n",
       " 'technique',\n",
       " 'techniques',\n",
       " 'techniques',\n",
       " 'technologies',\n",
       " 'tens',\n",
       " 'termed',\n",
       " 'tertiary',\n",
       " 'text',\n",
       " 'that',\n",
       " 'that',\n",
       " 'that',\n",
       " 'that',\n",
       " 'that',\n",
       " 'that',\n",
       " 'that',\n",
       " 'that',\n",
       " 'that',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'the',\n",
       " 'their',\n",
       " 'their',\n",
       " 'there',\n",
       " 'there',\n",
       " 'there',\n",
       " 'there',\n",
       " 'these',\n",
       " 'these',\n",
       " 'these',\n",
       " 'these',\n",
       " 'these',\n",
       " 'these',\n",
       " 'they',\n",
       " 'this',\n",
       " 'this',\n",
       " 'this',\n",
       " 'this',\n",
       " 'this',\n",
       " 'this',\n",
       " 'this',\n",
       " 'this',\n",
       " 'this',\n",
       " 'though',\n",
       " 'though',\n",
       " 'thousands',\n",
       " 'threading',\n",
       " 'three',\n",
       " 'three',\n",
       " 'three',\n",
       " 'throughput',\n",
       " 'throughput',\n",
       " 'time',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'to',\n",
       " 'together',\n",
       " 'tools',\n",
       " 'tools',\n",
       " 'topologically',\n",
       " 'towards',\n",
       " 'transduction',\n",
       " 'transporting',\n",
       " 'type',\n",
       " 'types',\n",
       " 'types',\n",
       " 'typically',\n",
       " 'understand',\n",
       " 'understand',\n",
       " 'understanding',\n",
       " 'unique',\n",
       " 'uniquely',\n",
       " 'unknown',\n",
       " 'use',\n",
       " 'used',\n",
       " 'used',\n",
       " 'used',\n",
       " 'used',\n",
       " 'used',\n",
       " 'usually',\n",
       " 'value',\n",
       " 'variety',\n",
       " 'various',\n",
       " 'vast',\n",
       " 'via',\n",
       " 'viable',\n",
       " 'virtual',\n",
       " 'virtually',\n",
       " 'virtually',\n",
       " 'visualize',\n",
       " 'vital',\n",
       " 'vocabularies',\n",
       " 'was',\n",
       " 'way',\n",
       " 'way',\n",
       " 'way',\n",
       " 'well',\n",
       " 'when',\n",
       " 'whether',\n",
       " 'which',\n",
       " 'which',\n",
       " 'which',\n",
       " 'which',\n",
       " 'which',\n",
       " 'whose',\n",
       " 'whose',\n",
       " 'widespread',\n",
       " 'with',\n",
       " 'with',\n",
       " 'with',\n",
       " 'within',\n",
       " 'without',\n",
       " 'work',\n",
       " 'work',\n",
       " 'x']"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(allwords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['3d',\n",
       " 'a',\n",
       " 'ability',\n",
       " 'about',\n",
       " 'accelerate',\n",
       " 'accuracy',\n",
       " 'acid',\n",
       " 'acyclic',\n",
       " 'added',\n",
       " 'aims',\n",
       " 'algorithms',\n",
       " 'all',\n",
       " 'also',\n",
       " 'although',\n",
       " 'amino',\n",
       " 'amounts',\n",
       " 'an',\n",
       " 'analyse',\n",
       " 'analysed',\n",
       " 'analyses',\n",
       " 'analysis',\n",
       " 'analyze',\n",
       " 'and',\n",
       " 'another',\n",
       " 'application',\n",
       " 'applications',\n",
       " 'are',\n",
       " 'area',\n",
       " 'artificial',\n",
       " 'as',\n",
       " 'associating',\n",
       " 'atoms',\n",
       " 'attempts',\n",
       " 'augment',\n",
       " 'automate',\n",
       " 'available',\n",
       " 'b',\n",
       " 'barcoding',\n",
       " 'based',\n",
       " 'be',\n",
       " 'becoming',\n",
       " 'been',\n",
       " 'behind',\n",
       " 'between',\n",
       " 'biodiversity',\n",
       " 'bioinformatic',\n",
       " 'bioinformatics',\n",
       " 'biological',\n",
       " 'biology',\n",
       " 'biomedical',\n",
       " 'bonds',\n",
       " 'both',\n",
       " 'bovine',\n",
       " 'branch',\n",
       " 'by',\n",
       " 'c',\n",
       " 'called',\n",
       " 'can',\n",
       " 'capture',\n",
       " 'cases',\n",
       " 'categorised',\n",
       " 'cell',\n",
       " 'cells',\n",
       " 'cellular',\n",
       " 'central',\n",
       " 'certain',\n",
       " 'challenges',\n",
       " 'chia',\n",
       " 'chromatin',\n",
       " 'chromosome',\n",
       " 'citation',\n",
       " 'classified',\n",
       " 'codes',\n",
       " 'collection',\n",
       " 'commonly',\n",
       " 'completely',\n",
       " 'complex',\n",
       " 'comprise',\n",
       " 'computational',\n",
       " 'computer',\n",
       " 'computers',\n",
       " 'concepts',\n",
       " 'condition',\n",
       " 'conformation',\n",
       " 'connected',\n",
       " 'connections',\n",
       " 'constructed',\n",
       " 'contain',\n",
       " 'content',\n",
       " 'controlled',\n",
       " 'could',\n",
       " 'course',\n",
       " 'covering',\n",
       " 'cow',\n",
       " 'crystallography',\n",
       " 'currently',\n",
       " 'cytometry',\n",
       " 'data',\n",
       " 'databases',\n",
       " 'de',\n",
       " 'deals',\n",
       " 'describe',\n",
       " 'describes',\n",
       " 'descriptions',\n",
       " 'designed',\n",
       " 'determine',\n",
       " 'determined',\n",
       " 'determines',\n",
       " 'developed',\n",
       " 'diagnostics',\n",
       " 'different',\n",
       " 'dimensional',\n",
       " 'directed',\n",
       " 'directly',\n",
       " 'disease',\n",
       " 'disjointed',\n",
       " 'dna',\n",
       " 'docking',\n",
       " 'domains',\n",
       " 'done',\n",
       " 'draws',\n",
       " 'drug',\n",
       " 'dynamic',\n",
       " 'easily',\n",
       " 'effort',\n",
       " 'efforts',\n",
       " 'empirical',\n",
       " 'employ',\n",
       " 'encephalopathy',\n",
       " 'encountered',\n",
       " 'entity',\n",
       " 'environment',\n",
       " 'enzymes',\n",
       " 'essential',\n",
       " 'every',\n",
       " 'evolution',\n",
       " 'evolutionary',\n",
       " 'example',\n",
       " 'examples',\n",
       " 'exceptions',\n",
       " 'exist',\n",
       " 'experiment',\n",
       " 'experimental',\n",
       " 'experiments',\n",
       " 'expression',\n",
       " 'far',\n",
       " 'field',\n",
       " 'fields',\n",
       " 'finding',\n",
       " 'flow',\n",
       " 'for',\n",
       " 'formation',\n",
       " 'forms',\n",
       " 'foundry',\n",
       " 'from',\n",
       " 'fully',\n",
       " 'function',\n",
       " 'functionally',\n",
       " 'fundamental',\n",
       " 'gain',\n",
       " 'gene',\n",
       " 'general',\n",
       " 'genes',\n",
       " 'genome',\n",
       " 'genomic',\n",
       " 'graphs',\n",
       " 'growing',\n",
       " 'growth',\n",
       " 'have',\n",
       " 'hemoglobin',\n",
       " 'heuristics',\n",
       " 'hi',\n",
       " 'high',\n",
       " 'holistic',\n",
       " 'homologous',\n",
       " 'homology',\n",
       " 'humans',\n",
       " 'ideas',\n",
       " 'identical',\n",
       " 'identification',\n",
       " 'if',\n",
       " 'image',\n",
       " 'imagery',\n",
       " 'images',\n",
       " 'imaging',\n",
       " 'important',\n",
       " 'impossible',\n",
       " 'improving',\n",
       " 'in',\n",
       " 'include',\n",
       " 'including',\n",
       " 'infer',\n",
       " 'informatics',\n",
       " 'information',\n",
       " 'integrate',\n",
       " 'integrated',\n",
       " 'interaction',\n",
       " 'interactions',\n",
       " 'into',\n",
       " 'involve',\n",
       " 'involves',\n",
       " 'is',\n",
       " 'it',\n",
       " 'its',\n",
       " 'k',\n",
       " 'key',\n",
       " 'knowledge',\n",
       " 'known',\n",
       " 'large',\n",
       " 'leghemoglobin',\n",
       " 'legumes',\n",
       " 'library',\n",
       " 'life',\n",
       " 'linguistics',\n",
       " 'literature',\n",
       " 'localization',\n",
       " 'location',\n",
       " 'loci',\n",
       " 'low',\n",
       " 'mad',\n",
       " 'magnetic',\n",
       " 'majority',\n",
       " 'make',\n",
       " 'makes',\n",
       " 'many',\n",
       " 'mapping',\n",
       " 'may',\n",
       " 'measurement',\n",
       " 'measurements',\n",
       " 'metabolic',\n",
       " 'metabolism',\n",
       " 'metabolites',\n",
       " 'methods',\n",
       " 'microbiome',\n",
       " 'mine',\n",
       " 'modeling',\n",
       " 'modelling',\n",
       " 'modern',\n",
       " 'molecular',\n",
       " 'molecule',\n",
       " 'molecules',\n",
       " 'more',\n",
       " 'most',\n",
       " 'movement',\n",
       " 'much',\n",
       " 'native',\n",
       " 'near',\n",
       " 'needed',\n",
       " 'network',\n",
       " 'networks',\n",
       " 'niche',\n",
       " 'nmr',\n",
       " 'not',\n",
       " 'notion',\n",
       " 'novo',\n",
       " 'nuclear',\n",
       " 'number',\n",
       " 'objectivity',\n",
       " 'obo',\n",
       " 'observer',\n",
       " 'obtained',\n",
       " 'of',\n",
       " 'often',\n",
       " 'on',\n",
       " 'once',\n",
       " 'one',\n",
       " 'only',\n",
       " 'ontologies',\n",
       " 'ontology',\n",
       " 'open',\n",
       " 'or',\n",
       " 'organised',\n",
       " 'organism',\n",
       " 'organization',\n",
       " 'other',\n",
       " 'others',\n",
       " 'oxygen',\n",
       " 'paper',\n",
       " 'particular',\n",
       " 'partitioning',\n",
       " 'parts',\n",
       " 'pathways',\n",
       " 'performing',\n",
       " 'pet',\n",
       " 'phenotypes',\n",
       " 'phylogenetics',\n",
       " 'physically',\n",
       " 'physics',\n",
       " 'populations',\n",
       " 'possible',\n",
       " 'practical',\n",
       " 'predict',\n",
       " 'predicted',\n",
       " 'predicting',\n",
       " 'prediction',\n",
       " 'predictions',\n",
       " 'primary',\n",
       " 'principle',\n",
       " 'prion',\n",
       " 'problem',\n",
       " 'processes',\n",
       " 'processing',\n",
       " 'protein',\n",
       " 'proteinligand',\n",
       " 'proteinpeptide',\n",
       " 'proteinprotein',\n",
       " 'proteins',\n",
       " 'provide',\n",
       " 'proximity',\n",
       " 'published',\n",
       " 'purpose',\n",
       " 'purposes',\n",
       " 'quantification',\n",
       " 'quaternary',\n",
       " 'question',\n",
       " 'ray',\n",
       " 'read',\n",
       " 'reflects',\n",
       " 'regulatory',\n",
       " 'relationships',\n",
       " 'relevant',\n",
       " 'reliably',\n",
       " 'remains',\n",
       " 'replace',\n",
       " 'research',\n",
       " 'resonance',\n",
       " 'resources',\n",
       " 'resulting',\n",
       " 'richness',\n",
       " 'rotatable',\n",
       " 's',\n",
       " 'same',\n",
       " 'scratch',\n",
       " 'secondary',\n",
       " 'seeks',\n",
       " 'seems',\n",
       " 'sequence',\n",
       " 'sequences',\n",
       " 'serve',\n",
       " 'set',\n",
       " 'shapes',\n",
       " 'share',\n",
       " 'signal',\n",
       " 'similar',\n",
       " 'simple',\n",
       " 'simulation',\n",
       " 'simulations',\n",
       " 'single',\n",
       " 'small',\n",
       " 'so',\n",
       " 'solution',\n",
       " 'some',\n",
       " 'space',\n",
       " 'spatial',\n",
       " 'species',\n",
       " 'spectroscopy',\n",
       " 'speed',\n",
       " 'spongiform',\n",
       " 'standardise',\n",
       " 'state',\n",
       " 'statistical',\n",
       " 'statistics',\n",
       " 'still',\n",
       " 'structural',\n",
       " 'structure',\n",
       " 'structures',\n",
       " 'studying',\n",
       " 'sub',\n",
       " 'subcellualr',\n",
       " 'subcellular',\n",
       " 'subsystems',\n",
       " 'such',\n",
       " 'system',\n",
       " 'systems',\n",
       " 'tackle',\n",
       " 'tads',\n",
       " 'taxonomic',\n",
       " 'technique',\n",
       " 'techniques',\n",
       " 'technologies',\n",
       " 'tens',\n",
       " 'termed',\n",
       " 'tertiary',\n",
       " 'text',\n",
       " 'that',\n",
       " 'the',\n",
       " 'their',\n",
       " 'there',\n",
       " 'these',\n",
       " 'they',\n",
       " 'this',\n",
       " 'though',\n",
       " 'thousands',\n",
       " 'threading',\n",
       " 'three',\n",
       " 'throughput',\n",
       " 'time',\n",
       " 'to',\n",
       " 'together',\n",
       " 'tools',\n",
       " 'topologically',\n",
       " 'towards',\n",
       " 'transduction',\n",
       " 'transporting',\n",
       " 'type',\n",
       " 'types',\n",
       " 'typically',\n",
       " 'understand',\n",
       " 'understanding',\n",
       " 'unique',\n",
       " 'uniquely',\n",
       " 'unknown',\n",
       " 'use',\n",
       " 'used',\n",
       " 'usually',\n",
       " 'value',\n",
       " 'variety',\n",
       " 'various',\n",
       " 'vast',\n",
       " 'via',\n",
       " 'viable',\n",
       " 'virtual',\n",
       " 'virtually',\n",
       " 'visualize',\n",
       " 'vital',\n",
       " 'vocabularies',\n",
       " 'was',\n",
       " 'way',\n",
       " 'well',\n",
       " 'when',\n",
       " 'whether',\n",
       " 'which',\n",
       " 'whose',\n",
       " 'widespread',\n",
       " 'with',\n",
       " 'within',\n",
       " 'without',\n",
       " 'work',\n",
       " 'x']"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(set(allwords))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
